lean4-infoview/src/infoview/rpcSessions.tsx

```diff
@@ -156,7 +161,16 @@ export class RpcSessions implements Disposable {
                 newRs.setSelf = rs.setSelf
                 return newRs
             })
+
+            return newSesh
         })
+        // NOTE(WN): because we insert into `#connecting` above, this conditional may only fail
+        // if the whole `newSesh` promise chain is executed immediately (if that's even possible).
+        // In that case we are either already connected or know we failed to do so, so don't want
+        // to mark as connecting.
+        if (this.#connecting.has(uri)) {
+            this.#connecting.set(uri, newSesh)
```

Comment:
Does the comment match the code?  You say "don't want to make as connecting" but then you go right ahead and set #connecting?

Reply:
"that case" is the else-branch afaict (where it is indeed marked as not connecting).